Dockerfile

```diff
@@ -5,6 +5,11 @@ FROM rust:1.66.1 AS builder
 RUN USER=root cargo new --bin trin
 WORKDIR /trin
 
+# Docker build command *MUST* include --build-arg GIT_HASH=... to set this value
+# eg. --build-arg GIT_HASH=$(git rev-parse HEAD)
+ARG GIT_HASH
```

Comment:
I think it is better if we set a default value for GIT_HASH:

```sh
ARG GIT_HASH=<default_value>
```

This will use the default value **if** `GIT_HASH` is not provided via `--build-arg`.

I'm not sure what this default value should be but I think we don't want to have this hard requirement to always provide a git hash when building the docker image. It is adding additional overhead over the typical use case for just building the docker image and not caring for the git hash.

Reply:
From what I can tell, it's not technically a hard requirement. W/o a default value, the arg is set to an empty string. But, it's not a huge deal to put a default value in there. I updated it to `GIT_HASH=unknown`. I've also updated the `build.rs` script to set the rustc env var `GIT_HASH` to an empty string rather than panicking. Then the actual util function will set the hash or "unknown" accordingly.